content/client-lib-development-guide/features.textile

```diff
@@ -696,3 +702,362 @@ The following default values are configured for the client library:
 * @(DF1)@ Realtime defaults:
 ** @(DF1a)@ @connectionStateTtl@ integer - default 60s. The duration that Ably will persist the connection state when a Realtime client is abruptly disconnected. When the client is in the @DISCONNECTED@ state, once this TTL has passed, the client should change the state to the @SUSPENDED@ state signifying that the state is now lost i.e. channels need to be reattached manually. Note that this default is overriden by @connectionStateTtl@, if specified in the @ConnectionDetails@ of the @CONNECTED@ @ProtocolMessage@
 ** @(DF1b)@ @realtimeRequestTimeout@ - default 10s. When a realtime client library is establishing a connection with Ably, or sending a @HEARTBEAT@, @CONNECT@, @ATTACH@, @DETACH@ or @CLOSE@ @ProtocolMessage@ to Ably, this is the amount of time that the client library will wait before considering that request as failed and triggering a suitable failure condition
+
+h2(#idl). Interface Definition
+
+The following bespoke IDL (Interface Definition Language) describes the types and classes present in the Rest and Realtime client libraries.
+
+Please note the following conventions:
+
+* Types are capitalized
+* Attribute and method names are lowercase
+* Attributes are denoted as @attributeName: Type@
+* Instance methods are denoted as @methodName(argName: Type, argName: Type) -> ReturnType@ (@ReturnType@ may be omitted)
+* Callback functions / closures are denoted as @(argName: Type, argName: Type) -> ReturnType@ (@ReturnType@ is usually omitted)
+* Values produced by I/O (e.g. a request) are prefixed with @=> io@. In some platforms (JS) those values are passed as arguments to a callback argument, instead of being returned. I/O always can fail, but how do they fail is undefined in the spec, so it's also undefined here
+* Enums are denoted as @.A | .B | .C@
+* @Type?@ denotes an optional type, i.e. a type that can be null plus all values of type @Type@
```

Comment:
> @Type?@ denotes an optional type

Wouldn't we say nullable?

I realise that what you're trying to say is that if an argument to a method has nullable type then  the library can equivalently make that an optional argument. But just saying the type is optional is too confusing.